src/pages/index.js

```diff
@@ -85,43 +127,157 @@ const addModal = new PopUpWithForm(
+  const cardElement = new Card(
+    data,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCard,
+    handleLike
+  );
   return cardElement.getView();
 }
 
 function renderCard(cardData) {
-  const addCard = new Card(cardData, cardSelector, handleImageClick);
-  return addCard.getView();
+  const cardInstance = new Card(
+    cardData,
+    cardSelector,
+    handleImageClick,
+    handleDeleteCard,
+    handleLike
+  );
+  return cardInstance.getView();
 }
 
 function handleAddCardSubmit(inputValues) {
-  const name = inputValues.name;
-  const link = inputValues.link;
-  const cardEl = renderCard({ name, link });
-  cardSection.addItem(cardEl);
-  addModal.close();
+  addModal.setLoading(true);
+  api
+    .addCard(inputValues)
+    .then((data) => {
+      const cardEl = renderCard(data);
+      section.addItem(cardEl);
+      addModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addModal.setLoading(false);
+    });
 }
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
-  },
-  ".cards__list"
-);
-cardSection.renderItems();
+function handleAvatarSubmit(url) {
+  profileAvatarPopUp.setLoading(true);
+  api
+    .updateAvatar(url)
+    .then((data) => {
+      userInfo.setAvatar(data);
+      profileAvatarPopUp.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileAvatarPopUp.setLoading(false);
+    });
+}
+
+function handleDeleteCard(cardInstance) {
+  cardDeletePopUp.open();
+  cardDeletePopUp.setSubmitAction(() => {
+    cardDeletePopUp.setLoading(true, "Deleting...");
+    api
+      .deleteCard(cardInstance._data._id)
+      .then(() => {
+        cardInstance.handleDeleteButton();
+        cardDeletePopUp.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        cardDeletePopUp.setLoading(false, "Yes");
+      });
+  });
+}
+
+function handleLike(cardInstance) {
+  if (cardInstance.isLiked) {
+    api
+      .dislikeCard(cardInstance.id)
+      .then((data) => {
+        cardInstance.setIsLiked(false);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .likeCard(cardInstance.id)
+      .then((data) => {
+        cardInstance.setIsLiked(true);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
+
+/*Delete card confirmation*/
+const deleteConfirmPopup = new PopUpWithConfirmation({
+  popUpSelector: "#delete-modal",
+});
+deleteConfirmPopup.setSubmitAction(() => {
+  deleteConfirmPopup.setLoading(true);
+  api
+    .deleteCard(cardId)
+    .then(() => {
+      cardElement.remove();
+      deleteConfirmPopup.close();
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    })
+    .finally(() => {
+      deleteConfirmPopup.setLoading(false);
+    });
```

Comment:
You've created 2 instances of `PopUpWithConfirmation`. Please, delete the extra one